File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis.VisualBasic/BoundContinueStatement.cs
using System.Diagnostics;
using Microsoft.CodeAnalysis.VisualBasic.Symbols;

namespace Microsoft.CodeAnalysis.VisualBasic
{
	internal sealed class BoundContinueStatement : BoundStatement
	{
		private readonly LabelSymbol _Label;

		public LabelSymbol Label => _Label;

		public BoundContinueStatement(SyntaxNode syntax, LabelSymbol label, bool hasErrors)
			: base(BoundKind.ContinueStatement, syntax, hasErrors)
		{
			_Label = label;
		}

		public BoundContinueStatement(SyntaxNode syntax, LabelSymbol label)
			: base(BoundKind.ContinueStatement, syntax)
		{
			_Label = label;
		}

		[DebuggerStepThrough]
		public override BoundNode Accept(BoundTreeVisitor visitor)
		{
			return visitor.VisitContinueStatement(this);
		}

		public BoundContinueStatement Update(LabelSymbol label)
		{
			if ((object)label != Label)
			{
				BoundContinueStatement boundContinueStatement = new BoundContinueStatement(base.Syntax, label, base.HasErrors);
				boundContinueStatement.CopyAttributes(this);
				return boundContinueStatement;
			}
			return this;
		}
	}
}

File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis/VisualBasic/BoundContinueStatement.vb
Imports Microsoft.CodeAnalysis
Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
Imports System
Imports System.Diagnostics

Namespace Microsoft.CodeAnalysis.VisualBasic
	Friend NotInheritable Class BoundContinueStatement
		Inherits BoundStatement
		Private ReadOnly _Label As LabelSymbol

		Public ReadOnly Property Label As LabelSymbol
			Get
				Return Me._Label
			End Get
		End Property

		Public Sub New(ByVal syntax As SyntaxNode, ByVal label As LabelSymbol, ByVal hasErrors As Boolean)
			MyBase.New(BoundKind.ContinueStatement, syntax, hasErrors)
			Me._Label = label
		End Sub

		Public Sub New(ByVal syntax As SyntaxNode, ByVal label As LabelSymbol)
			MyBase.New(BoundKind.ContinueStatement, syntax)
			Me._Label = label
		End Sub

		<DebuggerStepThrough>
		Public Overrides Function Accept(ByVal visitor As BoundTreeVisitor) As BoundNode
			Return visitor.VisitContinueStatement(Me)
		End Function

		Public Function Update(ByVal label As LabelSymbol) As Microsoft.CodeAnalysis.VisualBasic.BoundContinueStatement
			Dim boundContinueStatement As Microsoft.CodeAnalysis.VisualBasic.BoundContinueStatement
			If (CObj(label) = CObj(Me.Label)) Then
				boundContinueStatement = Me
			Else
				Dim boundContinueStatement1 As Microsoft.CodeAnalysis.VisualBasic.BoundContinueStatement = New Microsoft.CodeAnalysis.VisualBasic.BoundContinueStatement(MyBase.Syntax, label, MyBase.HasErrors)
				boundContinueStatement1.CopyAttributes(Me)
				boundContinueStatement = boundContinueStatement1
			End If
			Return boundContinueStatement
		End Function
	End Class
End Namespace